mlir/lib/Dialect/MIOpen/Transforms/ConvToGemmLowering.cpp

```diff
@@ -195,10 +195,20 @@ LogicalResult backwardWeightAtomicAdd(Conv2DBwdWeightOp op,
   auto stridesAttr = op->template getAttrOfType<ArrayAttr>("strides");
   auto paddingAttr = op->template getAttrOfType<ArrayAttr>("padding");
 
+  auto xdlopsV2Attr = op->template getAttrOfType<BoolAttr>("xdlopsV2");
+  bool isXdlops = (xdlopsV2Attr && xdlopsV2Attr.getValue() == true);
+
   // Get shape of filter tensor.
   auto filterType = op.filter().getType().template cast<MemRefType>();
   auto filterShape = filterType.getShape();
 
+  // Determine whether to use workspace.
+  bool hasWorkspace =
+      (filterType.getElementType() == b.getF16Type() && isXdlops);
+  if (hasWorkspace) {
+    assert(op.workspace());
```

Comment:
assert() just run in debug, is there right?

Reply:
Will do.